### DISCUSSION TITLE
Mod still needs an update: Broken-ish

### DISCUSSION THREAD

**Discussion by CuriousRiz** [TitanNano/TS4ControlAnySim#56] [Q&A] [answered]
I know this mod is outdated but it was still working up until the most recent patch. On Wednesday I was playing with a Scenario household and my teen sim died from embarrassment. So I reloaded my save, so I could try saving my sim from death but then my computer crashed after I loaded up the save. After everything was working again, I turned the game back on, loaded the save and I couldn't click any of my sims. The sim pie menu wouldn't show up but it did for everything else. I did lots of things and finally 50/50'ed my Mods and ControlAnySim was the cause. Which was weird because everything was fine up until the crash. So for now it's been removed from my game. I hope you find the time to update this soon. If not, maybe you could ask someone to take over for you temporarily.

**Comment by keyofrassilon** [comment, marked as answer]
Delete the json file located in the control any sim folder, wherever you have it located. You should be able to load the game again. The mod isnt broken it just doesnt error trap that very well. Sims no longer in the world will cause an access violation when they are "selected" because they no longer exist as selectable. The mod has always had this problem. It will probably never beccome outdated unless EA decides to rewrite the python code for selecting sims

**Reply by TitanNano** [reply]
Please feel free to open an issue with a full exception attached, and I will look into it.

**Reply by TitanNano** [reply]
Any issue with adding sims to the skewer should be caught and the sim in question should then be ignored. So, I'm curious what exactly is going wrong.

@CuriousRiz if you still have a broken json file, please attach it to a comment. You can attach the log file as well.

**Reply by CuriousRiz** [reply]
I did delete the json file and I think the log file (just in case) and it started to work again. Now I think I still have the json and log file on my desktop but I don't think I'll be able to attach it here until next week. I wish I saw this earlier.

**Reply by TitanNano** [reply]
No problem. If you happen to find the files somewhere (on your desktop, in the trash bin, wherever), just attach them to a comment whenever you want.

**Reply by CuriousRiz** [reply]
[ControlAnySim JSON & LOG files.zip](https://github.com/TitanNano/TS4ControlAnySim/files/9946492/ControlAnySim.JSON.LOG.files.zip)
I was able to do a quick look and found the files today. Thank you for your time. I really appreciate it. Gotta go now and Have a Wonderful Day!

**Reply by TitanNano** [reply]
Ok, so your JSON file just contains a bunch of garbage data, which broke the mod. The good news is that I just implemented a fix for such a case and the mod will now clean up such a broken file. The fix will be part of the next version. Thanks for providing the files.

**Reply by keyofrassilon** [reply]
I'm wondering if you should just go ahead and add the ability to select sims while on vacation since it's also been an issue for a while. Some ppl have requested this so I added it & those using my mod have that ability, it extends some of the capabilities of ControlAnySim inc other features. I did the work for you already although you might find a better way of doing this. Reference below:

[make_sim_selectable](https://github.com/keyofrassilon/Sims4ScriptCore/blob/1ade60e20378abd9198839a760e2b4a7e093a056/Scripts/scripts_core/sc_jobs.py#L2242)

[make_sim_unselectable](https://github.com/keyofrassilon/Sims4ScriptCore/blob/1ade60e20378abd9198839a760e2b4a7e093a056/Scripts/scripts_core/sc_jobs.py#L2270)

**Reply by TitanNano** [reply]
As a first step, could you open a New issue and describe what is going wrong in the current Version of the mod? I understand this is a bug that occurs when you try to make Sims selectable while on vacation?
